File: branches/Medusa/Source/Fantasy.Jobs/Management/JobStartupPriorityFilter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Fantasy.Jobs.Management
{
    public class JobStartupPriorityFilter : IJobStartupFilter 
    {

        #region IJobStartupFilter Members

        public IEnumerable<JobMetaData> Filter(IEnumerable<JobMetaData> source)
        {
            return source.OrderByDescending(j => j.Priority); 
        }

        #endregion
    }
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/management/JobStartupPriorityFilter.java
package fantasy.jobs.management;

import fantasy.collections.*;

public class JobStartupPriorityFilter implements IJobStartupFilter
{


	public final Iterable<JobMetaData> Filter(Iterable<JobMetaData> source)
	{
		
		return new Enumerable<JobMetaData>(source).orderByDescending(new Selector<JobMetaData, Integer>(){

			@Override
			public Integer select(JobMetaData item) {
				return item.getPriority();
			}});

	}

}